### ISSUE TITLE
Flake8 is not reporting all the errors in a file

### ISSUE THREAD

**Issue by HemaSowjanyaMamidi** [PyCQA/flake8#1971]
### how did you install flake8?

```console
$ pip install flake8
```

### unmodified output of `flake8 --bug-report`

```json
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.10.12",
    "system": "Linux"
  },
  "plugins": [
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.11.1"
    },
    {
      "plugin": "pyflakes",
      "version": "3.2.0"
    }
  ],
  "version": "7.0.0"
}
```

#### what I expected to happen

Report all the errors in a given file

#### sample code

```python
import os
import math

def find_smallest(arr):
    smallest = arr[0]
    smallest_index = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
        smallest = arr[i]
            smallest_index = i
    return smallest_index


def selection_sort(arr):
    new_arr = []
    for i in range(len(arr)):
        smallest = find_smallest(arr)
        new_arr.append(arr.pop(smallest))
    reurn new_arr


array = [100, 5, 72, 41, 80, 1, 99, 36, 27, 78]

print(selection_sort(array))  # [1, 5, 27, 36, 41, 72, 78, 80, 99, 100]

```

#### commands ran

```console
$ flake8 sample.py
...
```

#### What happened

Only the indentation error was reported. Rest of the errors such as unused imports, invalid syntax are not reported.

**Comment by asottile** [comment]
a syntax error will prevent the ast parsing so it cannot report any other errors
